### PR TITLE
Add a callback parameter that is called once collection is initialised

### PR DESCRIPTION
**Pull request by ahoppen**
This adds a callback parameter to the constructor and `refresh` method that is called once initialisation of the collection from localforage has finished / the data reloaded from localforage. 

This allows the application to wait with further collection updates until the collection has been properly initialised, eliminating potential data races.

I believe, the following To-Do in [ethereum/meteor-dapp-wallet](https://github.com/ethereum/meteor-dapp-wallet) can be resolved when this PR is merged: https://github.com/ethereum/meteor-dapp-wallet. I might give it a shot when I have time.

### CODE REVIEW

**Comment by ashish979** [comment]
Hi @ahoppen 

I have around 12 collections that I am persisting on indexDB(localforage). How can I wait for the data to load from localforage to localcollection so that I can make use of the helper methods defined on client side meteor code.

**Comment by ahoppen** [comment]
TLDR: Use my fork of the package and wait for all collections to finish loading.

## Install my fork of the package

Since this PR is not merged yet, you would need to manually use my version of the package. For this just add a directory names `packages` in the root of your meteor project, clone my [fork](https://github.com/ahoppen/meteor-persistent-minimongo2) there, check out the branch `initialisation-callback` and rename the created directory to `frozeman/persistent-minimongo2`. Depending on your platform, getting the slash into the directory name may turn out to be tricky. I somehow managed to do it on Mac by renaming the directory from the terminal.

Afterwards, meteor should use the local package instead. Deleting the directory from packages should switch you back to the official distribution.

## Wait for all collections to load
Create a promise for every `PersistentCollection` you create that gets fulfilled when the collection is initialised and then just use `Promise.all` to wait for all promises to get fulfilled.

Here is a code snippet:
```javascript
const collections = ...;
const yourAppName = ...;

function initialiseCollections(initialisationCallback) {
    const initialisationPromises = [];

    for (const collection of collections) {
        const promise = new Promise((resolve, reject) => {
            new PersistentMinimongo2(collection, yourAppName, () => {
                resolve();
            });
        });
        initialisationPromises.push(promise);
    }

    Promise.all(initialisationPromises).then(() => {
        initialisationCallback();
    });
}
```

I hope that answers your question.

**Review by frozeman — APPROVED** [review]

**Comment by frozeman** [comment]
Sorry for the long wait :( it takes some time for me to go through my inbox nowadays :)

**Comment by frozeman** [comment]
ive published a new version: 0.4.0
